tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ResponseTransform.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public class ResponseTransform
+    {
+        
+        /// <summary>
+        /// Base class used to describe transforming a played back response with necessary 
+        /// changes directly from a request.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="response"></param>
+        public virtual void ApplyTransform(HttpRequest request, HttpResponse response)
+        {
+            // empty by default
+        }
+    }
```

Comment:
```suggestion
    public abstract class ResponseTransform
    {
        
        /// <summary>
        /// Base class used to describe transforming a played back response with necessary 
        /// changes directly from a request.
        /// </summary>
        /// <param name="request"></param>
        /// <param name="response"></param>
        public abstract void ApplyTransform(HttpRequest request, HttpResponse response);
    }
```

If every subclass should implement this method (which I assume it should), then an abstract class/method is better.